src/vmm/src/device_manager/persist.rs

```diff
@@ -277,8 +277,23 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                     .map_err(Error::Legacy)?;
 
                     dev_manager
-                        .register_mmio_serial(vm, serial, Some(state.mmio_slot.clone()))
+                        .register_mmio_serial(vm, serial.clone(), Some(state.mmio_slot.clone()))
                         .map_err(Error::DeviceManager)?;
+
+                    // When restoring from a previously saved state, there is no serial
+                    // driver initialization, therefore the RDA (Received Data Available)
+                    // interrupt is not enabled. Because of that, the driver won't get
+                    // notified of any bytes that we send to the guest. The clean solution
+                    // would be to save the whole serial device state when we do the vm
+                    // serialization. For now we set that bit manually
+                    serial
+                        .lock()
+                        .expect("Poisoned lock")
+                        .serial
+                        .write(1, 0b0000_0001)
```

Comment:
I'd add a constant to represent that this 1 value is the interrupt enable register. Seems quite magical otherwise